### ISSUE TITLE
Problem in the Installation of CellBender

### ISSUE THREAD

**Issue by danielbarhanin** [broadinstitute/CellBender#93]
Hey!
Following the instrucation for installing CellBender, I got this following error:
ERROR: CellBender is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with bzr+http, bzr+https, bzr+ssh, bzr+sftp, bzr+ftp, bzr+lp, bzr+file, git+http, git+https, git+ssh, git+git, git+file, hg+file, hg+http, hg+https, hg+ssh, hg+static-http, svn+ssh, svn+http, svn+https, svn+svn, svn+file).

This happens when i did the command:
pip install -e CellBender

So i tried to install the package like it's instructed above with this command:
pip install git+https://github.com/broadinstitute/CellBender.git

But when I tried to run cellbender remove-background I get the following error:
Traceback (most recent call last):
  File "/usr/local/bin/cellbender", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/cellbender/base_cli.py", line 90, in main
    parser = get_populated_argparser()
  File "/usr/local/lib/python3.7/site-packages/cellbender/base_cli.py", line 76, in get_populated_argparser
    module_argparse = importlib.import_module('.'.join(module_argparse_str_list))
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'cellbender.remove_background'

Is there a solution for that? or new instructions for the installation?

Thanks in advance!

**Comment by sjfleming** [comment]
I think the problem is likely that `pip install -e CellBender` might not be pointing to your local cloned repository.

Can you try the following?
```bash
git clone https://github.com/broadinstitute/CellBender.git CellBender
```
then double-check that when you do
```bash
ls CellBender
```
you should be able to see the file `setup.py` in that directory.  If you can see that file, then
```bash
pip install -e CellBender
```
should work.

Alternatively, you can navigate into the directory where CellBender's `setup.py` file is located, and then do
```bash
pip install -e .
```

**Comment by bbimber** [comment]
@sjfleming We're also interested in this. Is there any reason why we should need to clone the repo and do a local install, as opposed to:
```
pip install git+https://github.com/broadinstitute/CellBender.git
```

The latter would be much nicer since package could declare a dependency on the github URL, as opposed to manually cloning and installing. I would have guessed that the pip install command above basically does a clone/install anyway.

Do you know a reason the command above would not work?

**Comment by bbimber** [comment]
A couple other observations:

If I manually clone the repo, do 'pip install -e', and then delete the local git copy, the tool fails. I would have expected pip install to copy the relevant code into a python library folder - is this not the case?

**Comment by mxposed** [comment]
@sjfleming please merge this https://github.com/broadinstitute/CellBender/pull/118 this would fix the issue

**Comment by sjfleming** [comment]
Thanks @mxposed , I will get that merged in before too long.

@bbimber , I think @mxposed 's PR will fix your first comment.  As for the other observation, I think that is actually expected behavior based on the `-e` flag I have included.  `-e` installs in an editable mode.  I like this because I like to make edits, but it's true that it doesn't always make sense.   :)
If you install without the `-e`, then it should (?) continue to work after you delete the local git repo.

**Comment by sjfleming** [comment]
Closed by #238
